src/android/LocationManager.java

```diff
@@ -745,7 +752,12 @@ public PluginResult run() {
 	}
     
     private void startMonitoringForRegion(final JSONObject arguments, final CallbackContext callbackContext) {
-        
+
+		if( !hasBlueToothPermission() ){
+			debugWarn("startMonitoringForRegion not permitted");
+			return;
+		}
+
```

Comment:
I think we need to call callbackContext.sendPluginResult(pluginResult);  where the PluginResult status is OK and pluginResult.setKeepCallback(true).  The reason is if we don't do this, the success callback in the JS side is never called.  It could make certain things hang.